### PR TITLE
Fix to get all port related attributes from config_db

### PR DESCRIPTION
**Pull request by paavaanan**
**- What I did**
- interface_naming_mode functions fetch the alias_interface_name via sonic-cfggen. But, with the recent change in config_db removes all port related entries when a sepecific interface is down. This affects the existing implementation of fetching port details to following fucntions [ interface_alias_to_name(), interface_name_to_allias(), InterfaceAliasConverter() ] and throws key error.

**- How I did it**

- To fix this, harnessed the existing port_config.ini file to fetch the interface details and populated a port_dict and this fetch the alias interface names quickly.

NOTE: config commands are entirely modified from linux commands to db format. This fix does not cover interface validation for those commands.

### CODE REVIEW

**Comment by paavaanan** [comment]
https://github.com/Azure/SONiC/issues/268

File: config/main.py [hunk 1]
```diff
@@ -12,16 +12,50 @@
...
+SONIC_CFGGEN_PATH = '/usr/local/bin/sonic-cfggen'
+HWSKU_KEY = 'DEVICE_METADATA.localhost.hwsku'
+PLATFORM_KEY = 'DEVICE_METADATA.localhost.platform'
 
 #
 # Helper functions
 #
 
+# Returns platform and HW SKU
+def get_platform_and_hwsku():
```
Comment (stcheng) [inline config/main.py]: since this script talks directly with configuration database, it is better to retrieve the information e.g. HWSKU and platform from the database rather than calling commands

File: show/main.py [hunk 2]
```diff
@@ -43,15 +49,43 @@ def read_config(self, filename):
         except configparser.NoSectionError:
             pass
 
+# Returns platform and HW SKU
+def get_platform_and_hwsku():
```
Comment (stcheng) [inline show/main.py]: same as above

File: show/main.py [hunk 3]
```diff
@@ -43,15 +49,43 @@ def read_config(self, filename):
...
         self.alias_max_length = 0
-        cmd = 'sonic-cfggen -d --var-json "PORT"'
-        p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
-        self.port_dict = json.loads(p.stdout.read())
+
+        # Get platform and hwsku
+        (platform, hwsku) = get_platform_and_hwsku()
+        port_config = os.path.join("/usr/share/sonic/device", platform, hwsku,
+                                    "port_config.ini")
+        (self.port_dict, _) = get_port_config(platform, hwsku, port_config)
```
Comment (stcheng) [inline show/main.py]: same as above

File: config/main.py [hunk 4]
```diff
@@ -12,16 +12,50 @@
...
+        raise OSError("Cannot detect platform")
+
+    return (platform, hwsku)
+
+
+# Get platform, hwsku and populate port_dict
+(platform, hwsku) = get_platform_and_hwsku()
+port_config = os.path.join("/usr/share/sonic/device", platform, hwsku,
+                            "port_config.ini")
+(port_dict, _) = get_port_config(platform, hwsku, port_config)
```
Comment (stcheng) [inline config/main.py]: port name and alias mapping is also stored in the configuration database, it is better to query in that way.

File: config/main.py (hunk 1 above)
Comment (paavaanan) [reply config/main.py]: 👍 I can use PORT attribute in config_db to fetch both default and alias interfaces of each port. So, _get_hwsku() and _get_platform() are not necessary.

File: show/main.py (hunk 2 above)
Comment (paavaanan) [reply show/main.py]: Addressed in the new commit.

File: show/main.py (hunk 3 above)
Comment (paavaanan) [reply show/main.py]: Addressd in the new commit

File: config/main.py (hunk 4 above)
Comment (paavaanan) [reply config/main.py]: Agreed @stcheng. Addressed in the new commit.

**Comment by stcheng** [comment]
@paavaanan this is a nice article about the git commit message: https://medium.com/@preslavrachev/what-s-with-the-50-72-rule-8a906f61f09c

File: config/main.py [hunk 5]
```diff
@@ -42,13 +45,13 @@ def run_command(command, display_cmd=False, ignore_error=False):
...
-    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
-
-    port_dict = json.loads(p.stdout.read())
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    port_dict = config_db.get_table('PORT')
 
     if interface_alias is not None:
+        if not port_dict:
+            click.echo("port_dict is None!")
```
Comment (jleveque) [inline config/main.py]: If port_dict is None, then the next line will crash. Should you do something else here in addition to printing a message in order to handle this gracefully?

File: config/main.py [hunk 6]
```diff
@@ -60,13 +63,13 @@ def interface_alias_to_name(interface_alias):
...
-    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
-
-    port_dict = json.loads(p.stdout.read())
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    port_dict = config_db.get_table('PORT')
 
     if interface_name is not None:
+        if not port_dict:
+            click.echo("port_dict is None!")
```
Comment (jleveque) [inline config/main.py]: Same concern as above.

File: show/main.py [hunk 7]
```diff
@@ -43,15 +49,18 @@ def read_config(self, filename):
...
-        cmd = 'sonic-cfggen -d --var-json "PORT"'
-        p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
-        self.port_dict = json.loads(p.stdout.read())
+
+        config_db = ConfigDBConnector()
+        config_db.connect()
+        self.port_dict = config_db.get_table('PORT')
+
+        if not self.port_dict:
+            click.echo("port_dict is None!")
```
Comment (jleveque) [inline show/main.py]: Same concern as above.

**Comment by paavaanan** [comment]
> @paavaanan this is a nice article about the git commit message: https://medium.com/@preslavrachev/what-s-with-the-50-72-rule-8a906f61f09c

Thanks shoutian

File: config/main.py [hunk 8]
```diff
@@ -12,17 +12,20 @@
...
+from portconfig import get_port_config
 
 import aaa
 import mlnx
 
-SONIC_CFGGEN_PATH = "sonic-cfggen"
+PLATFORM_ROOT_PATH = '/usr/share/sonic/device'
+SONIC_CFGGEN_PATH = '/usr/local/bin/sonic-cfggen'
+HWSKU_KEY = 'DEVICE_METADATA.localhost.hwsku'
+PLATFORM_KEY = 'DEVICE_METADATA.localhost.platform'
```
Comment (jleveque) [inline config/main.py]: Why did you add definitions for `PLATFORM_ROOT_PATH`, `HWSKU_KEY` and `PLATFORM_KEY`? They are not used anywhere in the file.

File: config/main.py [hunk 9]
```diff
@@ -12,17 +12,20 @@
 from swsssdk import ConfigDBConnector
 from natsort import natsorted
 from minigraph import parse_device_desc_xml
+from portconfig import get_port_config
```
Comment (jleveque) [inline config/main.py]: Why add this import if you are not using it?

File: show/main.py [hunk 10]
```diff
@@ -15,9 +15,15 @@
...
 from swsssdk import ConfigDBConnector
 from swsssdk import SonicV2Connector
+from portconfig import get_port_config
 
 import mlnx
 
+PLATFORM_ROOT_PATH = '/usr/share/sonic/device'
+SONIC_CFGGEN_PATH = '/usr/local/bin/sonic-cfggen'
+HWSKU_KEY = 'DEVICE_METADATA.localhost.hwsku'
+PLATFORM_KEY = 'DEVICE_METADATA.localhost.platform'
```
Comment (jleveque) [inline show/main.py]: Why add these four constants? They are not used anywhere in the file.

File: show/main.py [hunk 11]
```diff
@@ -15,9 +15,15 @@
 import sonic_platform
 from swsssdk import ConfigDBConnector
 from swsssdk import SonicV2Connector
+from portconfig import get_port_config
```
Comment (jleveque) [inline show/main.py]: Why add this import if you are not using it?

File: config/main.py (hunk 8 above)
Comment (paavaanan) [reply config/main.py]: Joe, it is added as part of the initiall pull request. This is not necessary. Will remove the lines

File: config/main.py (hunk 9 above)
Comment (paavaanan) [reply config/main.py]: removed in the new commit

File: show/main.py (hunk 10 above)
Comment (paavaanan) [reply show/main.py]: Other than SONIC_CFGGEN_PATH removed unused constants

File: show/main.py (hunk 11 above)
Comment (paavaanan) [reply show/main.py]: removed in new commit

Comment (jleveque) [reply show/main.py]: ? This was not removed.

Comment (paavaanan) [reply show/main.py]: Addressed in new commit!

**Comment by stcheng** [comment]
thanks for fixing that!
